src/plots/plot.ts

```diff
@@ -451,27 +456,42 @@ export class Plot extends Component {
    * @return {Plots.PlotEntity[]}
    */
   public entities(datasets = this.datasets()): Plots.PlotEntity[] {
-    var entities: Plots.PlotEntity[] = [];
+    var entities = this._lightweightEntities(datasets);
+
+    entities.forEach((entity) => {
+      entity.selection = entity.drawer.selectionForIndex(entity.validDatumIndex);
+      delete(entity.drawer);
+      delete(entity.validDatumIndex);
```

Comment:
Why does entity.selection have to be evaluated here?  Since `selectionForIndex` caches, why can't we just fold `_lightweightEntities` into `entities`?